pallets/parachain-staking/src/types.rs

```diff
@@ -0,0 +1,1673 @@
+// Copyright 2020-2022 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Types for parachain-staking
+
+use crate::{
+    set::OrderedSet, BalanceOf, BottomDelegations, CandidateInfo, Config, DelegatorState, Error,
+    Event, Pallet, Round, RoundIndex, TopDelegations, Total, COLLATOR_LOCK_ID, DELEGATOR_LOCK_ID,
+};
+use frame_support::{
+    pallet_prelude::*,
+    traits::{tokens::WithdrawReasons, LockableCurrency},
+};
+use parity_scale_codec::{Decode, Encode};
+use sp_runtime::{
+    traits::{AtLeast32BitUnsigned, Saturating, Zero},
+    Perbill, Percent, RuntimeDebug,
+};
+use sp_std::{cmp::Ordering, collections::btree_map::BTreeMap, prelude::*};
+
+pub struct CountedDelegations<T: Config> {
+    pub uncounted_stake: BalanceOf<T>,
+    pub rewardable_delegations: Vec<Bond<T::AccountId, BalanceOf<T>>>,
+}
+
+#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct Bond<AccountId, Balance> {
+    pub owner: AccountId,
+    pub amount: Balance,
+}
+
+impl<A: Decode, B: Default> Default for Bond<A, B> {
+    fn default() -> Bond<A, B> {
+        Bond {
+            owner: A::decode(&mut sp_runtime::traits::TrailingZeroInput::zeroes())
+                .expect("infinite length input; no invalid inputs for type; qed"),
+            amount: B::default(),
+        }
+    }
+}
+
+impl<A, B: Default> Bond<A, B> {
+    pub fn from_owner(owner: A) -> Self {
+        Bond {
+            owner,
+            amount: B::default(),
+        }
+    }
+}
+
+impl<AccountId: Ord, Balance> Eq for Bond<AccountId, Balance> {}
+
+impl<AccountId: Ord, Balance> Ord for Bond<AccountId, Balance> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.owner.cmp(&other.owner)
```

Comment:
Not sure why not to compare `self.amount`.

Reply:
they made the decision to sort Bonds by owner address, not by balance for some reason, yeah.
Not sure why too, especially because Manta/pallets/parachain-staking/src/lib.rs:1697 then needs to order by stake.